app/src/main/java/com/fbada/projectone/domain/Car.kt

```diff
@@ -0,0 +1,572 @@
+    year: Int,
+    colour: AvailableColours,
+    fuelType: AvailableFuelTypes,
+    gearbox: AvailableGearBoxes
+) : BMW(dashboard, satNav, location, year, colour, fuelType, gearbox) {
+    /**
+     * Calculates a score based on dashboard type
+     */
+    fun dashboardScore(): Int {
+        return when (dashboard) {
+            AvailableBMWDashboards.ANALOGUE -> LOW
+            AvailableBMWDashboards.CONNECTEDDRIVE, AvailableBMWDashboards.IDRIVE -> MEDIUM
+            AvailableBMWDashboards.DIGITAL -> HIGH
+            AvailableBMWDashboards.HUD, AvailableBMWDashboards.SPORT -> ULTIMATE
+        }
+    }
+
+    /**
+     * Calculates a score based on sat nav type
+     */
+    fun satNavScore(): Int {
+        return when (satNav) {
+            AvailableSatNavs.GARMIN -> LOW
+            AvailableSatNavs.TOMTOM, AvailableSatNavs.MAGELLAN -> MEDIUM
+            AvailableSatNavs.NAVMAN, AvailableSatNavs.WAZE, AvailableSatNavs.GOOGLE -> HIGH
+            AvailableSatNavs.APPLE, AvailableSatNavs.HERE -> ULTIMATE
+        }
+    }
+
+    /**
+     * Calculates a combined navigation score based on dashboard and sat nav types and whether power steering is available.
+     */
+    fun alphaNavigationRatingBonus(): Int {
+        val combinedRatingScore = dashboardScore() + satNavScore()
+        return if (powerSteering) {
+            combinedRatingScore * 2
+        } else {
+            combinedRatingScore
+        }
+    }
+
+    /**
+     * Subclass Beta Model type car.  Extends [BMW] class.
+     * @property insurance whether insurance available
+     * @property insuranceType types of insurance
+     * @property powerSteering whether power steering available
+     */
+    abstract class Beta(
+        var insurance: Boolean,
+        var insuranceType: AvailableInsurance?,
+        private var powerSteering: Boolean,
+        dashboard: AvailableBMWDashboards,
+        satNav: AvailableSatNavs,
+        location: String,
+        year: Int,
+        colour: AvailableColours,
+        fuelType: AvailableFuelTypes,
+        gearbox: AvailableGearBoxes
+    ) : BMW(dashboard, satNav, location, year, colour, fuelType, gearbox) {
+        /**
+         * Calculates an insurance score based on insurance type
+         */
+        fun insuranceScore(): Int {
+            return if (insurance) {
+                when (insuranceType) {
+                    AvailableInsurance.BASIC_LIABILITY -> MIN
+                    AvailableInsurance.STANDARD_PROTECTION -> LOW
+                    AvailableInsurance.FULL_COVERAGE -> MEDIUM
+                    AvailableInsurance.PREMIUM_PROTECTION -> HIGH
+                    AvailableInsurance.ROADSIDE_ASSISTANCE -> ULTIMATE
+                    else -> {
+                        throw IllegalStateException("No insurance selected")
+                    }
+                }
+            } else {
+                throw IllegalStateException("Insurance Not Available")
+            }
+        }
```

Comment:
Proposal, can we simplify it to:
```
fun insuranceScore(): Int {
            return insuranceType?.takeIf { insurance }?.let {
                when (it) {
                    AvailableInsurance.BASIC_LIABILITY -> MIN
                    AvailableInsurance.STANDARD_PROTECTION -> LOW
                    AvailableInsurance.FULL_COVERAGE -> MEDIUM
                    AvailableInsurance.PREMIUM_PROTECTION -> HIGH
                    AvailableInsurance.ROADSIDE_ASSISTANCE -> ULTIMATE
                }
            } ?: throw IllegalStateException("Insurance Not Available")
        }
```
Also could you please rename `insurance` to `insuranceAvailable`, or something that would make more sense.
P.S. it feels like `AvailableInsurance` class is more an `InsuranceType` class(naming thing).